AK/SourceGenerator.h

```diff
@@ -94,6 +94,13 @@ class SourceGenerator {
         return SourceGenerator { m_builder, m_mapping, m_opening, m_closing, suffix };
     }
 
+    SourceGenerator fork_function(StringView return_type, StringView name, StringView arguments)
```

Comment:
With this signature we can't use the method to generate const functions. Let's have another parameter `Constness` (I hope you're familiar with the single-purpose bool-like enums). Also, let's rename return_type to something more generic, as that string can also be used for attributes like ALWAYS_INLINE.